esx_service/vmci/vmci_client.c

```diff
@@ -280,17 +285,28 @@ vsock_get_reply(be_sock_id *s, be_request *r, be_answer* a)
    ret = recv(s->sock_id, &b, sizeof b, 0);
    if (ret == -1 || ret != sizeof b) {
       fprintf(stderr, "Failed to receive len: ret %d (%s)\n", ret, strerror(errno));
+      if (a->errBuf) {
+         snprintf(a->errBuf, 500, "Failed to receive len : ret %d (%s)\n", ret, strerror(errno));
+      }
       return CONN_FAILURE;
    }
 
    a->buf = calloc(1, b);
    if (!a->buf) {
+      if (a->errBuf) {
+         snprintf(a->errBuf, 500, "Failed to allocate memory len : %d\n", b);
+      }
       return CONN_FAILURE;
    }
 
    ret = recv(s->sock_id, a->buf, b, 0);
    if (ret == -1 || ret != b) {
       free(a->buf);
+      a->buf = NULL;
+      fprintf(stderr, "Failed to receive message data: received %d expected %d\n", ret, b);
+      if (a->errBuf) {
+         snprintf(a->errBuf, 500, "Failed to receive message data: received %d expected %d\n", ret, b);
```

Comment:
Fixed, removed the fprintf's as there isn't a way to print anywhere except via a buffer that the go code can use.